service/worker/scanner/executions/workflows.go

```diff
@@ -120,16 +123,47 @@ type (
 	// ScanReportError is a type that is used to send either error or report on a channel.
 	// Exactly one of Report and ErrorStr should be non-nil.
 	ScanReportError struct {
-		Report   *common.ShardScanReport
+		Reports  []common.ShardScanReport
 		ErrorStr *string
 	}
 
 	// FixReportError is a type that is used to send either error or report on a channel.
 	// Exactly one of Report and ErrorStr should be non-nil.
 	FixReportError struct {
-		Report   *common.ShardFixReport
+		Reports  []common.ShardFixReport
 		ErrorStr *string
 	}
+
+	// PaginatedShardQueryRequest is the request used for queries which return results over all shards
+	PaginatedShardQueryRequest struct {
+		// StartingShardID is the first shard to start iteration from.
+		// Setting to nil will start iteration from the beginning of the shards.
+		StartingShardID *int
+		// LimitShards indicates the maximum number of results that can be returned.
+		// If nil or larger than allowed maximum, will default to maximum allowed.
+		LimitShards *int
+	}
+
+	// ShardQueryPaginationToken is used to return information used to make the next query
+	ShardQueryPaginationToken struct {
+		// NextShardID is one greater than the highest shard returned in the current query.
+		// NextShardID is nil if IsDone is true.
+		// It is possible to get NextShardID != nil and on the next call to get an empty result with IsDone = true.
+		NextShardID *int
+		IsDone      bool
```

Comment:
It seems like I only need to check `NextShardID != nil` to see if I need to keep calling? Then do we still need the `IsDone` field?

Reply:
This code doesn't use IsDone but I liked having IsDone = true for users manually querying the workflow.

Reply:
I see. Makes sense for end users.